app/Http/Controllers/SponsorLoginController.php

```diff
@@ -0,0 +1,142 @@
+        $email = (string) $request->input('email');
+
+        // Check if domain is valid and sponsor is active; if not, return JSON error
+        if (! $this->isValidSponsorDomain($email)) {
+            return $this->errorResponse(
+                'Authentication Error',
+                'Could not validate email or sponsor is no longer active. '.
+                'Contact hello@robojackets.org if the issue persists.'
+            );
+        }
+
+        // Get sponsor user and check if exists in one query
+        $sponsorUser = SponsorUser::where('email', $email)->first();
+        if (! $sponsorUser) {
+            // Create new SponsorUser model for new users
+            $sponsorUser = new SponsorUser();
+            $sponsorUser->email = $email;
+            $sponsorUser->save();
+        }
+
+        // Generate and dispatch OTP using Spatie
+        $sponsorUser->sendOneTimePassword();
+
+        // Cache minimal state for OTP verification
+        session([
+            'sponsor_email_pending' => $email,
+        ]);
+
+        return response()->json([
+            'success' => true,
+            'message' => 'One-Time Password Sent! Please type the one-time password sent to your email.',
+        ], 200);
+    }
+
+    public function verifyOtp(Request $request): JsonResponse
+    {
+        // Laravel will automatically throw an error if OTP is invalid
+        $request->validate([
+            'otp' => ['required', 'string', 'digits:6'],
+        ]);
+
+        // Validate session and retrieve user
+        $email = session('sponsor_email_pending');
+        if (! is_string($email) || $email === '') {
+            return $this->errorResponse(
+                'Session Expired',
+                'Your session has expired. Please start the login process again.'
+            );
+        }
+
+        // Retrieve existing user for OTP verification
+        // sole() ensures exactly one user exists (throws exception if 0 or >1 found)
+        $sponsorUser = SponsorUser::where('email', $email)->sole();
+
+        // Verify sponsor domain is still valid and active BEFORE verifying OTP
+        if (! $this->isValidSponsorDomain($email)) {
+            return $this->errorResponse(
+                'Authentication Error',
+                'Could not validate email or sponsor is no longer active. '.
+                'Please contact hello@robojackets.org if the issue persists.'
+            );
+        }
+
+        // Verify OTP using Spatie
+        $otp = (string) $request->input('otp');
+        $result = $sponsorUser->attemptLoginUsingOneTimePassword($otp);
+        if (! $result->isOk()) {
+            return $this->errorResponse('Invalid OTP', $result->validationMessage());
+        }
+
+        // Save user after successful OTP verification
+        $sponsorUser->save();
+
+        // Establish authenticated session using Laravel's Auth facade
+        // $request->session()->regenerate();
```

Comment:
`regenerate()` is already called inside `Auth::login()` so we don't need to call it again here: https://github.com/laravel/framework/blob/a6d9e4bc9bb09dca0ccc6efdf348c6aca5f71766/src/Illuminate/Auth/SessionGuard.php#L578

Remove this commented line